File: Year 2/Semester 4/Programming and Projection Enironments/Week 9/ProtobuffProjectJava/Persistance/src/main/java/gRPCProject/persistance/repository/IShowRepository.java
package gRPCProject.persistance.repository;

import gRPCProject.domain.Entity;
import gRPCProject.domain.ShowDTO;

import java.io.Serializable;

public interface IShowRepository <ID extends Serializable, E extends Entity<ID>> extends IRepository<ID, E> {
    Iterable<ShowDTO> getAllShowsDTO();
}

File: Year 2/Semester 4/Programming and Projection Environments/Week 9/ProtobuffProjectCSharp/persistence/interfaces/IShowRepository.cs
using ProtobuffProject.domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProtobuffProject.repository.interfaces
{
    public interface IShowRepository : IRepository<int, Show<int>>
    {
        List<ShowDTO<int>> GetAllShows();
    }
}
